proxyd/backend.go

```diff
@@ -593,47 +593,93 @@ func (b *BackendGroup) Forward(ctx context.Context, rpcReqs []*RPCReq, isBatch b
 
 	backends := b.Backends
 
-	// When `consensus_aware` is set to `true`, the backend group acts as a load balancer
-	// serving traffic from any backend that agrees in the consensus group
+	type indexedReqRes struct {
```

Comment:
Gotcha. I'll make each request in the batch to be individually handled, and if they are out of bounds it will simply resolve to an individual RPCErr without preventing the other requests to go thru.